src/app/commands/cmd_options.cpp

```diff
@@ -625,31 +447,249 @@ class OptionsWindow : public app::gen::Options {
     uninstallExtension()->Click.connect([this]{ onUninstallExtension(); });
     openExtensionFolder()->Click.connect([this]{ onOpenExtensionFolder(); });
 
+    // Reset checkboxes
+    
+    // Prevent the user from clicking "Reset" if they don't have anything selected.
+    auto validateYesButton = [this] {
+      resetSelectedButton()->setEnabled(
+        defaultReset()->isSelected() || installedReset()->isSelected() ||
+        recentReset()->isSelected() || perfileReset()->isSelected()
+      );
+    };
+    defaultReset()->Click.connect(validateYesButton);
+    installedReset()->Click.connect(validateYesButton);
+    recentReset()->Click.connect(validateYesButton);
+    perfileReset()->Click.connect(validateYesButton);
+    resetSelectedButton()->Click.connect([this] { onResetDefault(); });
+
+    defaultReset()->setSelected(true);
+
     // Apply button
     buttonApply()->Click.connect([this]{ onApply(); });
+    sectionListbox()->selectIndex(m_curSection);
+
+    // Refill languages combobox when extensions are enabled/disabled
+    m_extLanguagesChanges = App::instance()->extensions().LanguagesChange.connect([this]{ refillLanguages(); });
+
+    // Reload themes when extensions are enabled/disabled
+    m_extThemesChanges =App::instance()->extensions().ThemesChange.connect([this]{ reloadThemes(); });
+
+    loadFromPreferences();
+  }
+
+  void loadFromPreferences()
+  {
+    // Default extension to save files
+    fillExtensionsCombobox(defaultExtension(),
+                           m_pref.saveFile.defaultExtension());
+    fillExtensionsCombobox(exportImageDefaultExtension(),
+                           m_pref.exportFile.imageDefaultExtension());
+    fillExtensionsCombobox(exportAnimationDefaultExtension(),
+                           m_pref.exportFile.animationDefaultExtension());
+    fillExtensionsCombobox(exportSpriteSheetDefaultExtension(),
+                           m_pref.spriteSheet.defaultExtension());
```

Comment:
Yeah, some stuff triggered on auto-save and I cleaned it up manually but in the wrong direction 😅, I'll do the reverts

Reply:
Thanks about this! Probably this is a luxury we can afford now that we don't have the pre-commit configured.

I'm thinking out loud now, but probably in the future, in case some refactor needs to move some code, we can do a full clang-format of the modified files (but this is just for the future).